modules/repository-s3/src/test/java/org/elasticsearch/repositories/s3/S3BlobStoreContainerTests.java

```diff
@@ -330,86 +362,93 @@ public void testExecuteMultipartUploadAborted() {
-            when(client.initiateMultipartUpload(any(InitiateMultipartUploadRequest.class))).thenReturn(initResult);
+            final CreateMultipartUploadResponse.Builder initResult = CreateMultipartUploadResponse.builder();
+            initResult.uploadId(uploadId);
+            when(client.createMultipartUpload(any(CreateMultipartUploadRequest.class))).thenReturn(initResult.build());
 
-            when(client.uploadPart(any(UploadPartRequest.class))).thenAnswer(invocationOnMock -> {
+            when(client.uploadPart(any(UploadPartRequest.class), any(RequestBody.class))).thenAnswer(invocationOnMock -> {
                 final UploadPartRequest request = (UploadPartRequest) invocationOnMock.getArguments()[0];
-                final UploadPartResult response = new UploadPartResult();
-                response.setPartNumber(request.getPartNumber());
-                response.setETag(randomAlphaOfLength(20));
-                return response;
+                final UploadPartResponse.Builder response = UploadPartResponse.builder();
+                response.eTag(randomAlphaOfLength(20));
+                return response.build();
             });
 
             // Fail the completion request
             when(client.completeMultipartUpload(any(CompleteMultipartUploadRequest.class))).thenThrow(exceptions.get(stage));
         }
 
         final ArgumentCaptor<AbortMultipartUploadRequest> argumentCaptor = ArgumentCaptor.forClass(AbortMultipartUploadRequest.class);
-        doNothing().when(client).abortMultipartUpload(argumentCaptor.capture());
+        when(client.abortMultipartUpload(argumentCaptor.capture())).thenReturn(AbortMultipartUploadResponse.builder().build());
 
         final IOException e = expectThrows(IOException.class, () -> {
             final S3BlobContainer blobContainer = new S3BlobContainer(BlobPath.EMPTY, blobStore);
             blobContainer.executeMultipartUpload(randomPurpose(), blobStore, blobName, new ByteArrayInputStream(new byte[0]), blobSize);
         });
 
         assertEquals("Unable to upload or copy object [" + blobName + "] using multipart upload", e.getMessage());
-        assertThat(e.getCause(), instanceOf(AmazonClientException.class));
+        assertThat(e.getCause(), instanceOf(S3Exception.class));
         assertEquals(exceptions.get(stage).getMessage(), e.getCause().getMessage());
 
         if (stage == 0) {
-            verify(client, times(1)).initiateMultipartUpload(any(InitiateMultipartUploadRequest.class));
-            verify(client, times(0)).uploadPart(any(UploadPartRequest.class));
+            verify(client, times(1)).createMultipartUpload(any(CreateMultipartUploadRequest.class));
+            verify(client, times(0)).uploadPart(any(UploadPartRequest.class), any(RequestBody.class));
             verify(client, times(0)).completeMultipartUpload(any(CompleteMultipartUploadRequest.class));
             verify(client, times(0)).abortMultipartUpload(any(AbortMultipartUploadRequest.class));
```

Comment:
it's hard to gauge from just the delta, but are these unit tests providing much value that the integration tests with the fixtures don't? Could be a burden to continue to maintain them if they're not providing specific value

Reply:
Yeah possibly or possibly not. I derive some comfort from being able to check the exact sequence of calls we make against the client separately from checking that the eventual API calls yield the expected results. But it is indeed something of a burden.